pyuvdata/uvdata/corr_fits_src/corr_fits.pyx

```diff
@@ -121,3 +124,124 @@ cpdef numpy.ndarray[ndim=1, dtype=numpy.float64_t] get_cable_len_diffs(
+              )
+            )
+
+  return khat
+
+
+cpdef get_khat(rho, sig1, sig2):
+    rho = np.ascontiguousarray(rho, dtype=np.float64)
+    squeeze = False
+    if rho.ndim == 1:
+      squeeze = True
+      rho = rho.reshape(1, -1)
+    sig1 =  np.ascontiguousarray(sig1, dtype=np.float64)
+    sig2 =  np.ascontiguousarray(sig2, dtype=np.float64)
+
+    khat = np.zeros((rho.shape[0], rho.shape[1]), dtype=np.float64)
+    khat = _compute_khat(
+        rho, sig1, sig2
+    )
+
+    if squeeze:
+      khat = khat.squeeze()
+    return khat
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef numpy.ndarray[ndim=2, dtype=numpy.float64_t[:, ::1]] _get_cheby_coeff(
+    numpy.ndarray[ndim=3, dtype=numpy.float64_t] rho_coeff,
+    numpy.ndarray[ndim=1, dtype=numpy.int64_t] sv_inds_right1,
+    numpy.ndarray[ndim=1, dtype=numpy.int64_t] sv_inds_right2,
+    numpy.ndarray[ndim=1, dtype=numpy.float64_t] ds1,
+    numpy.ndarray[ndim=1, dtype=numpy.float64_t] ds2
+):
+  cdef int i
+  cdef int j
+  cdef int n = len(ds1)
+  cdef numpy.ndarray[ndim=2, dtype=numpy.float64_t] t = np.zeros((n, 3), dtype=np.float64)
+
+  for i in cython.parallel.prange(n, nogil=True):
+      for j in range(3):
+          t[i, j] = 1e4 * (
+              rho_coeff[(sv_inds_right1[i] - 1), (sv_inds_right2[i] - 1), j] * ds1[i] * ds2[i]
+              + rho_coeff[(sv_inds_right1[i] - 1), sv_inds_right2[i], j] * ds1[i] * (0.01 - ds2[i])
+              + rho_coeff[sv_inds_right1[i], (sv_inds_right2[i] - 1), j] * (0.01 - ds1[i]) * ds2[i]
+              + rho_coeff[sv_inds_right1[i], sv_inds_right2[i], j] * (0.01 - ds1[i]) * (0.01 - ds2[i])
+          )
+
+  return t
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cpdef numpy.ndarray[ndim=1, dtype=numpy.complex128_t[::1]] van_vleck_cheby(
+    numpy.ndarray[ndim=1, dtype=numpy.complex128_t] khat,
+    numpy.ndarray[ndim=3, dtype=numpy.float64_t] rho_coeff,
+    numpy.ndarray[ndim=1, dtype=numpy.int64_t] sv_inds_right1,
+    numpy.ndarray[ndim=1, dtype=numpy.int64_t] sv_inds_right2,
+    numpy.ndarray[ndim=1, dtype=numpy.float64_t] ds1,
+    numpy.ndarray[ndim=1, dtype=numpy.float64_t] ds2
+):
+
+  cdef numpy.float64_t[:, ::1] t = _get_cheby_coeff(rho_coeff, sv_inds_right1, sv_inds_right2, ds1, ds2)
+  cdef numpy.float64_t[:, ::1] k = np.array([khat.real, khat.imag])
+  cdef int n = k.shape[1]
+  cdef int i
+  cdef int j
+  cdef numpy.ndarray[ndim=1, dtype=numpy.complex128_t] rho = np.zeros((n), dtype=np.complex128)
+
+  for i in cython.parallel.prange(n, nogil=True):
+        rho[i] = (
+            k[0, i] * (t[i, 0] - 3 * t[i, 1] + 5 * t[i, 2])
+            + k[0, i] ** 3 * (4 * t[i, 1] - 20 * t[i, 2])
+            + k[0, i] ** 5 * (16 * t[i, 2]) + 1j * (
+            k[1, i] * (t[i, 0] - 3 * t[i, 1] + 5 * t[i, 2])
+            + k[1, i] ** 3 * (4 * t[i, 1] - 20 * t[i, 2])
+            + k[1, i] ** 5 * (16 * t[i, 2]))
+        )
+
+  return rho
```

Comment:
Do you happen to have an annotated compiled version of this from the cython interpreter? Always good to check we are doing the best we can in terms of cython -> C conversoins

Reply:
Here's a picture of the annotated compiled version, I can't upload the html file.
![annotated_corr_fits_pyx](https://user-images.githubusercontent.com/20100812/95261225-e15c7880-07de-11eb-9fa5-485957710555.png)

Reply:
I'll work on a notebook showing the time benefits.

Reply:
I'm going to jump into my last statement about this. There is a small python interaction on line 241. from the use of complex `1j`, it may be necessary to `cdef complex some_variable = 1j` for the python to go fully away? Maybe worth tweaking slightly. no reason to have a python line in there when we don't need it.